source/jsonnet/england-wales/lib/transforms.libsonnet

```diff
@@ -1,53 +1,92 @@
-    first_name: {
-      source: 'answers',
-      identifier: 'first-name',
-    },
-    middle_names: {
-      source: 'answers',
-      identifier: 'middle-names',
-    },
-    last_name: {
-      source: 'answers',
-      identifier: 'last-name',
-    },
+local firstNameFirstListItem(listName) = {
+  source: 'answers',
+  identifier: 'first-name',
+  list_item_selector: {
+    id: listName,
+    id_selector: 'first',
+    source: 'list',
   },
 };
 
-local formatSecondPersonName = {
-  transform: 'format_name',
-  arguments: {
-    include_middle_names: { source: 'previous_transform' },
-    first_name: {
-      source: 'answers',
-      identifier: 'first-name',
-      list_item_selector: {
-        source: 'location',
-        id: 'to_list_item_id',
-      },
-    },
-    middle_names: {
-      source: 'answers',
-      identifier: 'middle-names',
-      list_item_selector: {
-        source: 'location',
-        id: 'to_list_item_id',
-      },
-    },
-    last_name: {
-      source: 'answers',
-      identifier: 'last-name',
-      list_item_selector: {
-        source: 'location',
-        id: 'to_list_item_id',
-      },
-    },
+local middleNamesFirstListItem(listName) = {
+  source: 'answers',
+  identifier: 'middle-names',
+  list_item_selector: {
+    id: listName,
+    id_selector: 'first',
+    source: 'list',
+  },
+};
+
+local lastNameFirstListItem(listName) = {
+  source: 'answers',
+  identifier: 'last-name',
+  list_item_selector: {
+    id: listName,
+    id_selector: 'first',
+    source: 'list',
+  },
+};
+
+local firstNameCurrentListItem = {
+  source: 'answers',
+  identifier: 'first-name',
+};
+
+local middleNamesCurrentListItem = {
+  source: 'answers',
+  identifier: 'middle-names',
+};
+
+local lastNameCurrentListItem = {
+  source: 'answers',
+  identifier: 'last-name',
+};
+
+local firstNameToListItem = {
+  source: 'answers',
+  identifier: 'first-name',
+  list_item_selector: {
+    source: 'location',
+    id: 'to_list_item_id',
+  },
+};
+
+local middleNamesToListItem = {
+  source: 'answers',
+  identifier: 'middle-names',
+  list_item_selector: {
+    source: 'location',
+    id: 'to_list_item_id',
   },
 };
 
+local lastNameToListItem = {
+  source: 'answers',
+  identifier: 'last-name',
+  list_item_selector: {
+    source: 'location',
+    id: 'to_list_item_id',
+  },
+};
+
+local formatPersonName(source='', listName='household') = (
+  local firstNameSource = if source == 'first_list_item' then firstNameFirstListItem(listName) else if source == 'to_list_item' then firstNameToListItem else firstNameCurrentListItem;
+  local middleNamesSource = if source == 'first_list_item' then middleNamesFirstListItem(listName) else if source == 'to_list_item' then middleNamesToListItem else middleNamesCurrentListItem;
+  local lastNameSource = if source == 'first_list_item' then lastNameFirstListItem(listName) else if source == 'to_list_item' then lastNameToListItem else lastNameCurrentListItem;
```

Comment:
Could we consolidate the various first/to/current methods into three (one for each part of the name), and do the logic in each of those instead of here. The `source` value and list name could be passed directly into those methods, so these lines could go and below would become something like:

```
first_name: firstNameSource(source, listName),
middle_names: middleNamesSource(source, listName),
last_name: lastNameSource(source, listName),
```

Reply:
Good shout, done in [`35aa2ad` (#186)](https://github.com/ONSdigital/eq-questionnaire-schemas/pull/186/commits/35aa2ad3f5e67da832dc32b71fbaab40249ffc37)